backend/routes/follows.js

```diff
@@ -0,0 +1,48 @@
+import express from "express";
+import { Follow } from "../models/follow.js";
+import { User } from "../models/user.js";
+import { Stock } from "../models/stock.js";
+
+const router = express.Router();
+
+router.post("/follow", async (req, res) => {
+    const ticker = req.body.ticker.toUpperCase();
+
+    const user = req.session.user;
+    const stock = await Stock.findOne({ where: { ticker } });
+
+    const followData = {
+        UserId: user.id,
+        StockId: stock.id
+    };
+
+    try {
+        const follow = await Follow.create(followData);
+        res.status(200).json({ follow });
+    } catch (error) {
+        res.status(409).json({ error: "You already follow this stock." });
```

Comment:
It seems like you are relying on an exception being thrown to infer the follow relationship already exists.
But an exception can happen for other reasons, so this is not a very reliable way of finding out.

You can query the DB in order to find out before creating the new record.